internal/datacoord/compaction_trigger.go

```diff
@@ -438,6 +433,11 @@ func (t *compactionTrigger) handleGlobalSignal(signal *compactionSignal) error {
 			log.Warn("compaction plan skipped due to handler full")
 			break
 		}
+		if !t.isChannelCheckpointHealthy(group.channelName) && !signal.isForce {
+			log.Warn("compaction plan skipped due to channel checkpoint lag", zap.String("channel", signal.channel))
+			return merr.WrapErrServiceInternal("channel checkpoint lag", signal.channel)
```

Comment:
1. Global signal `m` uses "part-channel" as key, might need a channel cache to avoid dup-calculation
2. Global signal will try to trigger all collection, other channels might be healthy, no need to return, continue is fine

Reply:
fixed